### PR TITLE
Added urban dictionary support

### PR DESCRIPTION
**Pull request by matthew-lake**

### CODE REVIEW

**Comment by jgat** [comment]
what do you do if you want to define a phrase with more than one word?

It looks like `!urban foo bar` will look up `foo` and `bar` separately, but not `foo bar`.
Maybe it makes sense to only support one phrase per command.

File: scripts/urban.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+							var response = "";
+							for (var j = 0; j < responses.length; j++) {
+								var word = list[j].toUpperCase();
+								var definition = robot.getUrbanDef(responses[j]);
+								var example = robot.getUrbanExample(responses[j]);
+								response += ">" + word + ": " + definition + "\r\n";
+								response += "> \t _" + example + "_ \r\n";
+							}
+
+							if (list.length > 5) {
```
Comment (TRManderson) [inline scripts/urban.js]: should probs be `> limit`

**Comment by matthew-lake** [comment]
good point - I'll probably change it so it supports multi-word entries but not looking up multiple words at once

File: scripts/urban.js (hunk 1 above)
Comment (matthew-lake) [inline scripts/urban.js]: I blame @JosephGarrone

File: scripts/urban.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+	robot.getUrban = function (res, list) {
+		var completed = 0;
+		var responses = [];
+		var limit = 5;
+
+		var max = list.length;
+		if (list.length > limit) {
+			max = limit;
+		}
```
Comment (jgat) [inline scripts/urban.js]: alternative: `max = Math.min(list.length, limit)`

**Pull request merged**